app/models/planning_application.rb

```diff
@@ -735,6 +736,30 @@ def replacement_document_validation_request
     replacement_document_validation_requests.order(:created_at).last
   end
 
+  def time_extension_request
+    time_extension_validation_requests.order(:created_at).last
+  end
+
+  def closed_time_extension_requests
+    time_extension_validation_requests.where(state: "closed")
+  end
+
+  def has_only_time_extension_requests?
+    validation_requests.length == time_extension_validation_requests.length
+  end
+
+  def open_time_extension_requests
+    time_extension_validation_requests.where(state: "open")
+  end
+
+  def closed_time_extension_request
+    time_extension_validation_requests.where(state: "closed").order(:created_at).last
```

Comment:
Maybe we can use `closed_time_extension_requests` in the definition of `closed_time_extension_request` which also makes the relationship between them more clear (I initially missed why there were two with such similar names at first glance.)

```suggestion
    closed_time_extension_requests.order(:created_at).last
```

Reply:
Although actually I can't tell — do we use `closed_time_extension_request` (singular) anywhere? I can only find the plural version.